jenkins/hourly_maintenance.py

```diff
@@ -114,12 +115,29 @@ def RemoveVolumes():
     return err
 
 
+def KillLoopingBash():
+    err = 0
+    bash_procs = subprocess.check_output(['pgrep', 'bash']).split()
+
+    clock_hz = os.sysconf(os.sysconf_names['SC_CLK_TCK'])
+    for pid in bash_procs:
+        # man 5 proc
+        with open('/proc/%s/stat' % pid) as f:
+            stat = f.read().split()
+        utime = int(stat[13]) / clock_hz
+        utime_minutes = utime / 60
+        if utime_minutes > 30:
```

Comment:
Some general way to monitor for runaway processes would be great in general -- I think a more general solution than just examining bash is a good idea.